packages/release-mastermind/src/contexts/pullRequests.ts

```diff
@@ -9,139 +1,292 @@
-        this.newVersion.semantic.prerelease
-          ? `-${this.newVersion.semantic.prerelease}`
-          : ''
-      }${
-        this.newVersion.semantic.build
-          ? `+${this.newVersion.semantic.build}`
-          : ''
-      }`
-      log(LoggingLevels.debug, `New Version is: ${this.newVersion.name}`)
-    }
-  }
+	context: PRContext
+	config: PullRequestConfig
+	constructor(
+		util: Utils,
+		runners: Runners,
+		configs: Config,
+		curContext: CurContext,
+		dryRun: boolean
+	) {
+		if (curContext.type !== "pr")
+			throw new LoggingDataClass(
+				LoggingLevels.error,
+				"Cannot construct without pr context"
+			)
+		super(util, runners, configs, curContext, dryRun)
+		this.context = curContext.context
+		if (!configs.pr)
+			throw new LoggingDataClass(
+				LoggingLevels.error,
+				"Cannot start without config"
+			)
+		this.config = configs.pr
+	}
+
+	/**
+	 * Parse the PR Context
+	 * @author IvanFon, TGTGamer, jbinda
+	 * @since 1.0.0
+	 */
+	static async parse(
+		utils: Utils,
+		config: Config,
+		context: Context
+	): Promise<PRContext | undefined> {
+		const pr = context.payload.pull_request
+		if (!pr) {
+			return
+		}
+
+		log(
+			LoggingLevels.debug,
+			`context.payload.pull_request: ` +
+				JSON.stringify(context.payload.pull_request)
+		)
+
+		const IDNumber = pr.number
+		const labels = await utils.parsingData.labels(pr.labels).catch((err) => {
+			log(LoggingLevels.error, `Error thrown while parsing labels: `, err)
+			throw err
+		})
+		const files: string[] = await utils.api.files
+			.list(IDNumber)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while listing files: `, err)
+				throw err
+			})
+
+		const changes: number = await utils.api.pullRequests
+			.changes(pr.additions, pr.deletions)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while handling changes: `, err)
+				throw err
+			})
+
+		const reviews: Reviews = await utils.api.pullRequests.reviews
+			.list(IDNumber)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while handling reviews: `, err)
+				throw err
+			})
+
+		const pendingReview: boolean = await utils.api.pullRequests.reviews
+			.pending(reviews.length, pr.requested_reviewers.length)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while handling reviews: `, err)
+				throw err
+			})
+
+		const requestedChanges: number = await utils.api.pullRequests.reviews
+			.requestedChanges(reviews)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while handling reviews: `, err)
+				throw err
+			})
+
+		const approved: number = await utils.api.pullRequests.reviews
+			.isApproved(reviews)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while handling reviews: `, err)
+				throw err
+			})
+
+		const currentVersion: Version = await utils.versioning
+			.parse(config, config.pr?.ref)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while parsing versioning: `, err)
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Videndum/action-masterminds/pullRequest?prid=7872005)
```suggestion
        log(LoggingLevels.error, "Error thrown while parsing versioning: ", err)
```